packages/common/src/resources/get-portal-thumbnail-url.ts

```diff
@@ -0,0 +1,26 @@
+import { IPortal } from "@esri/arcgis-rest-portal";
+
+/**
+ * Construct a the full url to a portal thumbnail
+ *
+ * - If the portal has a thumbnail, construct the full url
+ * - If the portal is not public, append on the token
+ * @param portalUrl
+ * @param portal
+ * @param token
+ * @returns
+ */
+export function getPortalThumbnailUrl(
+  portalUrl: string,
+  portal: IPortal,
+  token?: string
+): string {
+  let thumbnailUrl = null;
+  if (portal?.thumbnail) {
+    thumbnailUrl = `${portalUrl}/portals/self/resources/${portal.thumbnail}`;
```

Comment:
I _think_ the `/self` is ok, but since you're passing in the portal, maybe use `/${portal.id}` as this would then work across orgs? Also consider constructing the portalUrl from the portal in here b/c it's got the info needed - `getPortalUrl(portal)` will do the work... this makes it a little nicer to use elsewhere